packages/nx/src/tasks-runner/task-graph-utils.ts

```diff
@@ -66,3 +70,42 @@ export function makeAcyclic(graph: {
     (t) => graph.dependencies[t].length === 0
   );
 }
+
+export function validateAtomizedTasks(
+  taskGraph: TaskGraph,
+  projectGraph: ProjectGraph
+): void {
+  if (process.env['NX_SKIP_ATOMIZER_VALIDATION']) {
+    return;
+  }
+  const tasksWithAtomizer = Object.values(taskGraph.tasks).filter(
+    (task) =>
+      projectGraph.nodes[task.target.project]?.data?.targets?.[
+        task.target.target
+      ]?.metadata?.nonAtomizedTarget !== undefined
```

Comment:
Mm, not sure I like this. `metadata` is not supposed to affect running of tasks :thinking: